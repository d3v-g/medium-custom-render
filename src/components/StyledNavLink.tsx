import { NavLink } from "react-router-dom"

const StyledNavLink: React.FC<{ path: string; label: string }> = ({
    path,
    label,
}) => (
    <NavLink
        to={path}
        className={({ isActive }) =>
            "w-20 rounded-lg border  py-2 shadow-lg " +
            (isActive
                ? "bg-red-500 dark:bg-red-900"
                : "bg-lime-900 dark:bg-green-900")
        }
    >
        {label}
    </NavLink>
)

export default StyledNavLink
