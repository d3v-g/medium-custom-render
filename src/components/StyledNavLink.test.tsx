import { render, screen } from "../test-utils"
import StyledNavLink from "./StyledNavLink"

describe("Styled navlink", () => {
    it("renders a link to the passed in path", () => {
        const path = "/test-path"
        const label = "Test label"
        render(<StyledNavLink path={path} label={label} />)
        const link = screen.getByRole("link")
        expect(link).toHaveAttribute("href", path)
    })
    it("renders the passed in label", () => {
        const path = "/test-path"
        const label = "Test label"
        render(<StyledNavLink path={path} label={label} />)
        const link = screen.getByRole("link")
        expect(link).toHaveTextContent(label)
    })

    it("renders active styles correctly", () => {
        const initialEntries = ["/test-path"]
        const path = "/test-path"
        const label = "Test label"
        const { container } = render(
            <StyledNavLink path={path} label={label} />,
            { initialEntries },
        )
        expect(container).toMatchSnapshot()
    })

    it("renders inactive styles correctly", () => {
        const initialEntries = ["/test-path"]
        const path = "/some-other-path"
        const label = "Test label"
        const { container } = render(
            <StyledNavLink path={path} label={label} />,
            { initialEntries },
        )
        expect(container).toMatchSnapshot()
    })
})
