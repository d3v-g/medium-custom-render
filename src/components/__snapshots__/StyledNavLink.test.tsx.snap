// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styled navlink > renders active styles correctly 1`] = `
<div>
  <a
    aria-current="page"
    class="w-20 rounded-lg border py-2 shadow-lg bg-red-500 dark:bg-red-900"
    href="/test-path"
  >
    Test label
  </a>
</div>
`;

exports[`Styled navlink > renders inactive styles correctly 1`] = `
<div>
  <a
    class="w-20 rounded-lg border py-2 shadow-lg bg-lime-900 dark:bg-green-900"
    href="/some-other-path"
  >
    Test label
  </a>
</div>
`;
